### ISSUE TITLE
Combobox selection not announced to screen reader users

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-components#4918]
### Summary
Selection is not announced to `combobox` in `selection-mode="single"` on-demand or when in focus. Other `selection-mode`s receive announcement when in focus, but not on-demand.

### Actual Behavior

After selecting a `combobox-item` where `selection-mode="single"` context is is not provided to the screen reader, even when the `combobox` is in focus and a selected value is displayed visually.

All `combobox`s do not receive context when a `combobox-item` value is selected.

#### JAWS transcript
```
Favorite fruit 
Choose a fruit
ExtendedSelect ListBox 
Not Selected Apple
To move to an item press the Arrow keys.
Not Selected Banana 
2 of 12
selected
```

### Expected Behavior
Provide context that an item has been selected in the combobox. A [combobox example](https://w3c.github.io/aria-practices/examples/combobox/combobox-select-only.html) that depicts an active value when the combobox is in focus.

- [x] If there is a selection in the combobox, it should be announced when in focused

#### JAWS transcript
```
Favorite Fruit Combo box 
Apple
To change the selection use the Arrow keys.
```
- [x] After selection of a combobox item, it should be announced.

#### JAWS transcript
```
Favorite Fruit Combo box 
Apple
To change the selection use the Arrow keys.
Favorite Fruit List box 
Apple 
2 of 13
To move to an item press the Arrow keys.
Banana 
3 of 13
Favorite Fruit Combo box 
Banana
To change the selection use the Arrow keys.
```

### Reproduction Sample

https://codepen.io/geospatialem/pen/VwXbRmB

### Reproduction Steps

1. Activate JAWS, NVDA, or VoiceOver
2. Navigate to the [sample](https://codepen.io/geospatialem/pen/VwXbRmB)
3. Expand the first `combobox` (single `selection-mode`)
4. Observe there is no announcement to the selected value.
5. Select a `combobox-item` from the list.
6. Observe there is no announcement provided when selecting a new value
7. Expand the second `combobox` (multi `selection-mode`)
8. Observe there is an announcement to the selected value.
9. Select a `combobox-item` from the list.
10. Observe there is no announcement provided when selecting a new value

### Reproduction Version

beta.83

### Relevant Info

Present across screen readers (JAWS, NVDA, VoiceOver)
cc #4388

#### 1. Potential solution to announce selection on-demand (applies to all `selection-mode` types)
- Add `aria-labelledby` to div for single selection
- Add `aria-live="polite"` for multi selection

#### 2. Potential solution to announce selection(s) for `selection-mode="single"`
- Add `aria-activedescendant` and `aria-controls` to the `div`
- Move the `onKeyDown` from the `Host` to the `div`
- Remove the `label id={`${labelUidPrefix}${guid}`}`
- In the `renderInput() add an id to the span with the `id={`${labelUidPrefix}${guid}`}`

#### Potential solution:
```ts
<Host>
        <div
          aria-activedescendant={`${listboxUidPrefix}${guid}`}
          aria-autocomplete="list"
          aria-controls={`${listboxUidPrefix}${guid}`}
          aria-expanded={toAriaBoolean(open || active)}
          aria-labelledby={`${labelUidPrefix}${guid}`}
          aria-live="polite"
          aria-owns={`${listboxUidPrefix}${guid}`}
          class={{
            wrapper: true,
            "wrapper--single": single || !this.selectedItems.length,
            "wrapper--active": open || active
          }}
          onClick={this.clickHandler}
          onKeyDown={this.keydownHandler}
          ref={this.setReferenceEl}
          role="combobox"
        >
```

#### Actions needed
- [x] Research

**Comment by driskull** [comment]
@geospatialem do you have any idea on how to fix this? We already have the `aria-expanded` set.

**Comment by geospatialem** [comment]
> @geospatialem do you have any idea on how to fix this? We already have the `aria-expanded` set.

I _think_ if some of the aria props are moved to the `Host` it'll mitigate the issue. I've got a similar fix in the works for `accordion-item`, and will link in the PR here after some follow-up testing tomorrow.

**Comment by benelan** [comment]
verified on `next` using NVDA (I don't have JAWS or VoiceOver but I assume Kitty already tested all of them)
